Dockerfile

```diff
@@ -1,17 +1,30 @@
-FROM golang:alpine as builder
+FROM golang:1.19.1-alpine as build
+
+ARG TARGETPLATFORM
+RUN echo "Building for ${TARGETPLATFORM}"
+
+WORKDIR /go/src
+
+COPY go.mod go.sum ./
+RUN go mod download
 
-WORKDIR /go/src/github.com/boynux/squid-exporter
 COPY . .
 
-# Compile the binary statically, so it can be run without libraries.
-RUN CGO_ENABLED=0 GOOS=linux go install -a -ldflags '-extldflags "-s -w -static"' .
+RUN CGO_ENABLED=0 go build -a -ldflags '-extldflags "-s -w -static"' -o /squid-exporter .
+
 
-FROM gcr.io/distroless/static:nonroot
-COPY --from=builder /go/bin/squid-exporter /usr/local/bin/squid-exporter
+FROM scratch as final
 
-# Allow /etc/hosts to be used for DNS
-COPY --from=builder /etc/nsswitch.conf /etc/nsswitch.conf
+LABEL org.opencontainers.image.title="Squid Exporter"
+LABEL org.opencontainers.image.description="This is a Docker image for Squid Prometheus Exporter."
+LABEL org.opencontainers.image.source="https://github.com/boynux/squid-exporter/"
+LABEL org.opencontainers.image.licenses="MIT"
+
+ENV SQUID_EXPORTER_LISTEN="0.0.0.0:9301"
```

Comment:
Since it's in docker, default listening on localhost doesn't work out of the box.